src/js/navIcons.js

```diff
@@ -148,4 +192,19 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
         icon.applier.change("isConfirmed", prevPanelNum === position && currentPanelNum > prevPanelNum);
     };
 
+    gpii.firstDiscovery.navIcons.saveVisitedPanelNum = function (that, currentPanelNum, prevPanelNum) {
+        var visitedPanelNums = that.model.visitedPanelNums;
+        if (currentPanelNum > prevPanelNum && $.inArray(prevPanelNum, visitedPanelNums) === -1 && $.inArray(prevPanelNum, that.options.iconHoles)) {
+            // Cannot fire the change request directly on visitedPanelNums due to http://issues.fluidproject.org/browse/FLUID-3504
+            that.applier.change("visitedPanelNums." + visitedPanelNums.length, prevPanelNum);
+        }
+    };
```

Comment:
I feel like this could be simplified if we had a model of all the panels and their state. 

```javascript
{0: "visited"}
```

or we saved the state in a array based on the panel index

```javascript
["visited"]
```